test/e2e/operator_test.go

```diff
@@ -876,3 +876,173 @@ func TestGatewayOwnership(t *testing.T) {
+	cfg := objcontour.Config{
+		Name:        contourName,
+		Namespace:   operatorNs,
+		RemoveNs:    true,
+		SpecNs:      specNs,
+		NetworkType: operatorv1alpha1.ClusterIPServicePublishingType,
+	}
+
+	cntr, err := newContour(ctx, kclient, cfg)
+	if err != nil {
+		t.Fatalf("failed to create contour %s/%s: %v", operatorNs, contourName, err)
+	}
+	t.Logf("created contour %s/%s", cntr.Namespace, cntr.Name)
+
+	// The contour should now report available.
+	if err := waitForContourStatusConditions(ctx, kclient, 3*time.Minute, contourName, operatorNs, expectedContourConditions...); err != nil {
+		t.Fatalf("failed to observe expected status conditions for contour %s/%s: %v", cfg.Namespace, cfg.Name, err)
+	}
+	t.Logf("observed expected status conditions for contour %s/%s", cfg.Namespace, cfg.Name)
+
+	// Create a sample workload for e2e testing.
+	appName := fmt.Sprintf("%s-%s", testAppName, testName)
+	if err := newDeployment(ctx, kclient, appName, cfg.SpecNs, testAppImage, testAppReplicas); err != nil {
+		t.Fatalf("failed to create deployment %s/%s: %v", cfg.SpecNs, appName, err)
+	}
+	t.Logf("created deployment %s/%s", cfg.SpecNs, appName)
+
+	// Wait for the sample workload to become available.
+	if err := waitForDeploymentStatusConditions(ctx, kclient, 3*time.Minute, appName, cfg.SpecNs, expectedDeploymentConditions...); err != nil {
+		t.Fatalf("failed to observe expected status conditions for deployment %s/%s: %v", cfg.SpecNs, appName, err)
+	}
+	t.Logf("observed expected status conditions for deployment %s/%s", cfg.SpecNs, appName)
+
+	if err := newClusterIPService(ctx, kclient, appName, cfg.SpecNs, 80, 8080); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", cfg.SpecNs, appName, err)
+	}
+	t.Logf("created service %s/%s", cfg.SpecNs, appName)
+
+	if err := newIngress(ctx, kclient, appName, cfg.SpecNs, appName, 80); err != nil {
+		t.Fatalf("failed to create ingress %s/%s: %v", specNs, appName, err)
+	}
+	t.Logf("created ingress %s/%s", specNs, appName)
+
+	// Create the client pod to test connectivity to the sample workload.
+	cliName := "test-client"
+	cliPod, err := newPod(ctx, kclient, specNs, cliName, "curlimages/curl:7.75.0", []string{"sleep", "600"})
+	if err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", specNs, cliName, err)
+	}
+	if err := waitForPodStatusConditions(ctx, kclient, 1*time.Minute, cliPod.Namespace, cliPod.Name, expectedPodConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions for pod %s/%s: %v", cliPod.Namespace, cliPod.Name, err)
+	}
+	t.Logf("observed expected status conditions for pod %s/%s", cliPod.Namespace, cliPod.Name)
+
+	// Get the Envoy ClusterIP to curl.
+	svcName := "envoy"
+	ip, err := envoyClusterIP(ctx, kclient, specNs, svcName)
+	if err != nil {
+		t.Fatalf("failed to get clusterIP for service %s/%s: %v", specNs, svcName, err)
+	}
+
+	// Curl the ingress from the client pod.
+	url := fmt.Sprintf("http://%s/", ip)
```

Comment:
How come this was done with curl in a pod rather than just from the test/go like other tests?

Reply:
@sunjayBhatia b/c the Contour is [configured](https://github.com/projectcontour/contour-operator/pull/316/files#diff-cf4b4e5424070a666a364d1d1b04011478d888d6d3d65c943ca7783333b7a6e4R914) with a "ClusterIPService" network publishing type. Therefore the Envoy Service is only reachable from within the cluster.

Reply:
ah gotcha, missed that 👍🏽